### PR TITLE
Added recipe for BayesTyper

### PR DESCRIPTION
**Pull request by micknudsen**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/bayestyper/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+source:
+  url: https://github.com/bioinformatics-centre/BayesTyper/archive/v{{ version }}.tar.gz
+  sha256: 917cd1b1ca7b5cfb6b8327138515ca1ad70878f0c4e8df393fddcbe42f281e14
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - boost
```
Comment (dpryan79) [inline recipes/bayestyper/meta.yaml]: Can you change this to `boost-cpp`? `boost` is actually a python interface that happens to pull an older `boost-cpp` in.

File: recipes/bayestyper/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  host:
+    - zlib
```
Comment (dpryan79) [inline recipes/bayestyper/meta.yaml]: Does this need zlib during the run time too or is it just statically compiling it in? In the latter case you can just put zlib in the build section.

File: recipes/bayestyper/meta.yaml (hunk 1 above)
Comment (micknudsen) [reply recipes/bayestyper/meta.yaml]: Done. Let's see how it goes.

File: recipes/bayestyper/meta.yaml (hunk 2 above)
Comment (micknudsen) [reply recipes/bayestyper/meta.yaml]: No idea. I have tried removing it. I know very little (that is, nothing) about C++. Just noticed that `zlib` always seems to be a requirement.

**Comment by dpryan79** [comment]
One last tweak to make the zlib run-time dependency explicit and this is good to go, thanks!

**Pull request merged**